src/ORM/Hierarchy/Hierarchy.php

```diff
@@ -879,11 +879,11 @@ public function getParent($filter = null)
             return null;
         }
         $baseClass = $this->owner->baseClass();
-        $idSQL = $this->owner->getSchema()->sqlColumnForField($baseClass, 'ID');
-        return DataObject::get_one($baseClass, [
-            [$idSQL => $parentID],
-            $filter
-        ]);
+        $list = DataObject::get($baseClass)->setUseCache(true);
+        if ($filter !== null) {
+            $list = $list->where($filter);
```

Comment:
The `$filter` arg would have gone in as raw SQL through `get_one()` via `get()->where()` so we have to keep that here in case someone is intentionally passing raw SQL in.